src/ewatercycle/esmvaltool/search.py

```diff
@@ -0,0 +1,228 @@
+"""Search ESGF for datasets with ESMValTool."""
+from itertools import compress
+from typing import Literal
+
+from esmvalcore.config import CFG
+from esmvalcore.dataset import Dataset
+
+
+def search_esgf(
+    activity: str,
+    experiment: str,
+    frequency: Literal["hr", "3hr", "day"],
+    variables: list[str],
+    extended_mip_tables: bool = False,
+    use_cache: bool = True,
+    verbose: bool = True,
+) -> dict[str, set[str]]:
+    """Search through ESGF for datasets fitting your requirements.
+
+    Examples:
+
+    To find all model ensembles which have the "pr", "tas", "rsdt" and "orog" variables
+    (the ones required for wflow), for the ssp585 scenario.
+
+    .. code-block:: python
+        from ewatercycle.esmvaltool.search import search esgf
+
+        search_esgf(
+            activity="ScenarioMIP",
+            experiment="ssp585",
+            frequency="day",
+            variables=["pr", "tas", "rsdt", "orog"],
+            extended_mip_tables=False,
+            use_cache=True,
+            verbose=False,
+        )
+
+    Gives something like:
+
+    .. code-block:: python
+        {
+            'MPI-ESM1-2-HR': {'r1i1p1f1', 'r2i1p1f1'},
+            'MPI-ESM1-2-LR': {'r10i1p1f1', 'r11i1p1f1', ...  'r9i1p1f1'},
+            'INM-CM4-8': {'r1i1p1f1'},
+            'MRI-ESM2-0': {'r1i1p1f1', 'r2i1p1f1', 'r3i1p1f1', 'r4i1p1f1', 'r5i1p1f1'},
+            'IPSL-CM6A-LR': {'r1i1p1f1'},
+            'GFDL-CM4': {'r1i1p1f1'},
+        }
+
```

Comment:
maybe add a short line to indicate that although a dataset might be in the MIP tables, the actual node hosting the data could be offline. Node status can be found here: https://aims2.llnl.gov/nodes